plugins/gcloud/lib/samson_gcloud/image_tagger.rb

```diff
@@ -15,11 +17,20 @@ def tag(deploy)
           next unless digest.match?(/(^|\/|\.)gcr.io\//) # gcr.io or https://gcr.io or region like asia.gcr.io
           base = digest.split('@').first
 
-          if deploy.stage.production?
-            tag_image('production', deploy, base, digest)
+          tags = []
```

Comment:
These tags are common for all the build in the deploy, Then we need to bring the whole 'tags' array formation out of the build loop.

Reply:
Each build would have a separate digest and base though, which are arguments passed to `tag_image`, so it makes sense to call that for each build?

Reply:
That's right.. but following code is common for all the builds.
```
  tags = []
  stage = deploy.stage
  if stage.production? && !stage.no_code_deployed?
    tags << PRODUCTION_TAG
  end
  if env_permalink = stage.environments&.first&.permalink
    tags << "env-#{env_permalink}"
 end
  tags << stage.permalink
 # From docker distro: https://tinyurl.com/ycgh7qsp
   tags.grep(/^[\w][\w.-]{0,127}$/)
```